endpoint/endpoint.go

```diff
@@ -145,3 +147,33 @@ func NewEndpointWithTTL(dnsName, recordType string, ttl TTL, targets ...string)
 func (e *Endpoint) String() string {
 	return fmt.Sprintf("%s %d IN %s %s", e.DNSName, e.RecordTTL, e.RecordType, e.Targets)
 }
+
+// DNSEndpointSpec defines the desired state
+type DNSEndpointSpec struct {
+	Endpoints []*Endpoint `json:"endpoints,omitempty"`
+}
+
+// DNSEndpointStatus defines the observed state
+type DNSEndpointStatus struct {
```

Comment:
Question: should we have part of the contract between such a CRD and external-dns that external-dns maintains status? We should have `observedGeneration` here.

Reply:
This is a good question, and mostly raises another off topic question about optimisation for all the sources in general. currently, all the sources does not seem to be using informer pattern. they just do `List` the object types from the kube client. It is so because the design of external-dns needs all the objects (providing Endpoint) to feed the planner, which then calculates what changes need to be done in DNS provider.

Coming to this source in particular, it also uses the `List` operation to get all the CRD objects for now. we could in future write an informer and do a watch, which would give us a local cache of these objects. In that scenario, having such a contract would be meaningful. Also `observedGeneration` in status field makes sense for a single CRD object, but here we are dealing with list of such objects, and i am unsure how meaningful that is to maintain it.

Reply:
Well - the a single instance of this resource is itself a list. I think `observedGeneration` makes sense as a status field. Otherwise, how do you know - as the controller or the user - that the controller has 'seen' a particular generation of the resource?

Reply:
@pmorie, currently there is nothing in status, so `observedGeneration` does not make sense for such an object. It would be unnecessary complexity. wdyt?

Reply:
@pmorie, now i have added the `ObservedGeneration` field in the `Status` according to the discussion. PTAL.